### PR TITLE
Ota

### PR DESCRIPTION
**Pull request by copy-ninja**
to test, you'll need to point to my repository in the last line of update.lua
https://api.github.com/repos/copy-ninja/AlarmPanel/releases/latest

### CODE REVIEW

File: source-code/init.lua [hunk 1]
```diff
@@ -11,20 +11,25 @@ if wifi.sta.getconfig() == "" then
...
-  if gpio.read(4) == gpio.LOW then
-    gpio.write(4, gpio.HIGH)
-  else
-    gpio.write(4, gpio.LOW)
-  end
+if gpio.read(4) == gpio.LOW then
+  gpio.write(4, gpio.HIGH)
+else
+  gpio.write(4, gpio.LOW)
+end
```
Comment (heythisisnate) [inline source-code/init.lua]: weird indentation here. Isn't this inside the function?

File: source-code/server.lua [hunk 2]
```diff
@@ -89,7 +94,17 @@ httpd_set("/status", function(request, response)
 end)
 
 httpd_set("/restart", function(request, response)
-  tmr.create():alarm(5000, tmr.ALARM_SINGLE, node.restart)
+  print("Heap: ", node.heap(), "HTTP: ", "Restart")
+  tmr.create():alarm(5000, tmr.ALARM_SINGLE, function() node.restart() end)
```
Comment (heythisisnate) [inline source-code/server.lua]: Instead of creating a new function, I think you can just pass the function `node.restart`. Any reason you changed that?

Comment (copy-ninja) [inline source-code/server.lua]: passing just node.restart works, but it restarts because it's in error

**Comment by copy-ninja** [comment]
Yea, http.get doesn't work. The payload was only 1kilobyte. 
The only way is to stream it into a file, then read the file line by line to find what we need

**Comment by copy-ninja** [comment]
the flow for update is: 
first pass -> 
downloads manifest, checks if there's version difference, build the manifest list, put it in "manifest" table into a file 

second pass -> 
loads the "manifest" table and get each files. The files from github is hosted in AWS, so most of them will have HTTP 302 redirection. 
Once all the files have been downloaded into .tmp file. we'll process them again by removing the HTTP response header in the .tmp file

File: source-code/server.lua (hunk 2 above)
Comment (heythisisnate) [inline source-code/server.lua]: oh. haha!

**Pull request merged**